### PR TITLE
feat(analyzer): add German PII recognizers (DE_*)

### PR DESCRIPTION
**Pull request by MvdB**
## Summary

Adds 9 new pattern-based recognizers for German-language PII, covering the most relevant entity types under DSGVO/BDSG:

| Entity | Description | Legal Basis | Validation |
|---|---|---|---|
| `DE_TAX_ID` | Steueridentifikationsnummer (11 digits) | § 139b AO | ISO 7064 Mod 11,10 checksum |
| `DE_TAX_NUMBER` | Steuernummer / ELSTER (13 digits + slash formats) | § 139a AO | Regex (state code range 01–16) |
| `DE_PASSPORT` | Reisepass (ICAO Doc 9303, 9 chars) | PassG § 4, PassV | Strict + relaxed ICAO charset |
| `DE_ID_CARD` | Personalausweis — nPA (9 ICAO chars) + old T+8 format | PAuswG, PAuswV | Regex |
| `DE_SOCIAL_SECURITY` | Rentenversicherungsnummer (12 chars) | § 147 SGB VI | DRV checksum algorithm |
| `DE_HEALTH_INSURANCE` | Krankenversicherungsnummer / KVNR (10 chars) | § 290 SGB V | GKV-Spitzenverband checksum |
| `DE_KFZ` | KFZ-Kennzeichen / vehicle registration plate | FZV § 8 | Regex (district + letter-id + digits) |
| `DE_HANDELSREGISTER` | Handelsregisternummer (HRA/HRB prefix) | §§ 9, 14 HGB | Regex |
| `DE_PLZ` | Postleitzahl / postal code (5 digits) | — | Regex (range 01001–99998), score=0.05 |

## Design decisions

- All recognizers follow the existing `PatternRecognizer` pattern and are placed in `predefined_recognizers/country_specific/germany/`
- All are **disabled by default** (`enabled: false`) following the project convention for country-specific recognizers
- Three recognizers with checksum algorithms (`DE_TAX_ID`, `DE_SOCIAL_SECURITY`, `DE_HEALTH_INSURANCE`) implement `validate_result()` returning `True`/`False` to promote/suppress matches — not `None`, to ensure invalid numbers are filtered out rather than kept at base score
- `DE_PLZ` has intentionally low base confidence (0.05) due to high false-positive risk; documented in docstring
- `DE_TAX_NUMBER` includes both ELSTER 13-digit and regional slash-format patterns with graduated confidence scores
- All recognizers use `global_regex_flags: 26` (IGNORECASE | MULTILINE | DOTALL) consistent with the rest of Presidio

## Test plan

- [x] 9 test files covering all recognizers (`presidio-analyzer/tests/test_de_*.py`)
- [x] 133 test cases — positive matches, negative matches, edge cases
- [x] Separate `validate_result()` tests for checksum-based recognizers
- [x] All tests pass locally: `python -m pytest presidio-analyzer/tests/test_de_*.py`
- [x] `ruff check` passes with no issues
- [x] `default_recognizers.yaml` entries follow the existing country-specific pattern
- [x] `docs/supported_entities.md` updated with Germany section
- [x] `CHANGELOG.md` entry added under `[unreleased]`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by MvdB** [comment]
@microsoft-github-policy-service agree

**Comment by MvdB** [comment]
I know there's https://github.com/microsoft/presidio/pull/1830 , but sadly it seems inactive.
German community is waiting for our own entities though.

**Review by MvdB — COMMENTED** [review]
Implemented suggested changes.

File: docs/recipes/german-language-support/spacy_en_de.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+nlp_engine_name: spacy
```
Comment (SharonHart) [inline docs/recipes/german-language-support/spacy_en_de.yaml:1]: I would keep this file outside of the package for now, we are investigating ways to make pre-baked receipes, 
also region specific. We can add it to docs/recipes for now, WDYT?

**Comment by SharonHart** [comment]
@MvdB any of the recognizers here might be relevant as well? 
https://github.com/microsoft/presidio/pull/1830

**Comment by MvdB** [comment]
@SharonHart  The telematik_id in #1830 was not implemented because its format seems too flexible and lacks a strict, publicly verifiable validation scheme leading to high risk of false positives.
